packages/runtime/container-runtime/src/opLifecycle/outbox.ts

```diff
@@ -259,18 +260,49 @@ export class Outbox {
 			true /* disableGroupedBatching */,
 			resubmittingBatchId,
 		);
-		this.flushInternal(
+		this.flushInternalMainBatch(
 			this.mainBatch,
 			false /* disableGroupedBatching */,
 			resubmittingBatchId,
 		);
 	}
 
+	/**
+	 * Flushes the main batch to the ordering service. If the batch is empty, it will create an empty grouped batch
+	 * if BatchId is present (resubmit flow) and the runtime should send.
+	 */
+	private flushInternalMainBatch(
```

Comment:
Maybe rename to `flushInternalEvenIfEmpty`

Reply:
Future / Side note - I feel like all the different behaviors should be captured as properties on the BatchManager, not by different arguments / implementations of the `flushInternal` function.

What do you think?